common/modules/original_owner_helper/src/lib.rs

```diff
@@ -14,7 +14,7 @@ pub trait OriginalOwnerHelperModule {
         let farm_token_id = farm_token_mapper.get_token_id();
 
         let mut opt_original_owner = None;
-        for payment in payments.into_iter() {
+        for payment in payments.iter() {
```

Comment:
Wasn't this working?

```suggestion
        for payment in &payments {
```

Reply:
No. Tried for a few, I get this error.

error[E0277]: `&ManagedRef<'_, <Self as ContractBase>::Api, ...>` is not an iterator
  --> common/modules/original_owner_helper/src/lib.rs:17:24
   |
17 |         for payment in &payments {
   |                        ^^^^^^^^^ `&ManagedRef<'_, <Self as ContractBase>::Api, ...>` is not an iterator
   |
   = help: the trait `Iterator` is not implemented for `&ManagedRef<'_, <Self as ContractBase>::Api, ...>`
   = note: required for `&ManagedRef<'_, <Self as ContractBase>::Api, ...>` to implement `IntoIterator